website/api_versioned_docs/version-1.1/blockchain-integration/contracts/verifier.md

```diff
@@ -31,38 +31,217 @@ In this example, the `IS_EVEN` zkVM program verifies that the number, `x`, is ev
 
-## Deployed Verifiers
+## Verifier Implementations
 
 All of our deployed verifier contracts implement the [IRiscZeroVerifier][IRiscZeroVerifier] interface.
-It is recommended that you use the [RiscZeroVerifierRouter][RiscZeroVerifierRouter.sol], described below.
 
-### Verifier Router
+We deploy a base verifier implementation, the emergency stop wrapper, and the router as part of our [version management design][version-management].
+It is recommended that most applications use [RiscZeroVerifierRouter](#verifier-router).
+In some cases, your application may wish to make calls directly to either the emergency stop wrapper, or the base implementation.
+
+:::tip
+It is recommended that you use the [RiscZeroVerifierRouter](#verifier-router)
+:::
+
+## Verifier Router
 
 Calls to `RiscZeroVerifierRouter.verify()` will be routed to the appropriate base verifier contract depending on which version of the zkVM was used to generate the receipt.
 By using the `RiscZeroVerifierRouter`, your contract can accept multiple types of receipts, including batch-verified receipts and receipts generated with future improvements to the zkVM and proofs system.
 
 RISC Zero deploys and manages the `RiscZeroVerifierRouter` contract listed below.
 This contract will have verifiers added for each release of the zkVM, and will have verifiers removed in the case of security vulnerabilities.
-You can find detailed information in the [version management design][VersionManagement], including information about how to manage your own copy of the smart contracts, if your application requires it.
+You can find detailed information in the [version management design][version-management], including information about how to manage your own copy of the smart contracts, if your application requires it.
 
-### Contract Addresses
+## Contract Addresses
+
+> NOTE: The tables below are generated from the [`deployment.toml`][deployment.toml] file in the risc0-ethereum repository.
+
+<!-- GENERATED CONTENT BEGIN -->
+
+### Ethereum Mainnet (1)
 
-:::info
-Contracts are not deployed for unreleased versions.
-You can use the [deployed contracts for a released version][doc-released-contracts].
-:::
+| Contract                                   | Address                                                              |
+| ------------------------------------------ | -------------------------------------------------------------------- |
+| [RiscZeroVerifierRouterRouter][router-src] | [`0x8EaB2D97Dfce405A1692a21b3ff3A172d593D319`][router-1-etherscan]   |
+| [RiscZeroGroth16Verifier][verifier-src]    | [`0x94A4684d6F7085C19138Bd4f9F3295fa9943C622`][verifier-1-etherscan] |
+| [RiscZeroVerifierEmergencyStop][estop-src] | [`0x08AeD6C108E500540a9544beF7a8B8a05E056e87`][estop-1-etherscan]    |
+| TimelockController                         | [`0x0b144E07A0826182B6b59788c34b32Bfa86Fb711`][timelock-1-etherscan] |
+
```

Comment:
personally would add `<br>` between sections to create a bit more breathing room and make it clearer which header goes with which table

Reply:
Good idea
https://github.com/risc0/risc0/pull/2334/commits/3aa12ad0ed48dce3dbfa0ec50139e269f0b2125b